### PR TITLE
[qtmozembed] Improve view size and rotation handling. Fixes JB#32226.

### PR DESCRIPTION
**Pull request by tworaz**
This adjust the qtmozembed layer to similar changes in EmbedLite. The
main goal is to improve rotation and size changing performance.

### CODE REVIEW

File: src/qopenglwebpage.h [hunk 1]
```diff
@@ -145,6 +148,7 @@ private Q_SLOTS:
     bool mThrottlePainting;
     mutable QMutex mReadyToPaintMutex;
     bool mReadyToPaint;
+    bool mRotating;
```
Comment (rainemak) [inline src/qopenglwebpage.h]: This should go to private class.

File: src/qopenglwebpage.h [hunk 2]
```diff
@@ -38,6 +38,7 @@ class QOpenGLWebPage : public QObject
     Q_PROPERTY(QWindow *window READ window WRITE setWindow NOTIFY windowChanged FINAL)
     Q_PROPERTY(bool throttlePainting READ throttlePainting WRITE setThrottlePainting NOTIFY throttlePaintingChanged FINAL)
     Q_PROPERTY(bool readyToPaint READ readyToPaint WRITE setReadyToPaint NOTIFY readyToPaintChanged FINAL)
+    Q_PROPERTY(bool rotating READ rotating NOTIFY rotatingChanged FINAL)
```
Comment (rainemak) [inline src/qopenglwebpage.h]: Would be also good for ```QuickMozView```.

File: src/qgraphicsmozview_p.cpp [hunk 3]
```diff
@@ -35,6 +35,40 @@
...
+        rotation = mozilla::ROTATION_180;
+        break;
+    default:
+        break;
+    }
+    return rotation;
+}
+
+bool IsLandscape(Qt::ScreenOrientation orientation)
+{
```
Comment (rainemak) [inline src/qgraphicsmozview_p.cpp]: Not big deal but I'd prefer ```return orientation == Qt::LandscapeOrientation || orientation == Qt::InvertedLandscapeOrientation;``` over switch-case.

File: src/qopenglwebpage.h (hunk 1 above)
Comment (tworaz) [inline src/qopenglwebpage.h]: Done.

File: src/qopenglwebpage.h (hunk 2 above)
Comment (tworaz) [inline src/qopenglwebpage.h]: Done.

File: src/qgraphicsmozview_p.cpp (hunk 3 above)
Comment (tworaz) [inline src/qgraphicsmozview_p.cpp]: Done.

**Comment by tworaz** [comment]
Note this depends on https://github.com/nemomobile-packages/gecko-dev/pull/53 and https://github.com/nemomobile-packages/gecko-dev/pull/52 to work properly.

**Comment by rainemak** [comment]
Obsoleted by https://git.merproject.org/mer-core/qtmozembed/merge_requests/6

**Pull request closed without merging**